src/app/clusters/basic/basic.cpp

```diff
@@ -83,26 +83,85 @@ class PlatformMgrDelegate : public DeviceLayer::PlatformManagerDelegate
 
-} // anonymous namespace
-
-void emberAfBasicClusterServerInitCallback(chip::EndpointId endpoint)
+class BasicAttrAccess : public app::AttributeAccessInterface
 {
-    EmberAfStatus status;
-
-    char nodeLabel[DeviceLayer::ConfigurationManager::kMaxNodeLabelLength + 1];
-    if (ConfigurationMgr().GetNodeLabel(nodeLabel, sizeof(nodeLabel)) == CHIP_NO_ERROR)
+public:
+    BasicAttrAccess() : app::AttributeAccessInterface(Optional<EndpointId>::Missing(), app::Clusters::Basic::Id) {}
+    CHIP_ERROR Read(const app::ConcreteReadAttributePath & aPath, app::AttributeValueEncoder & aEncoder) override
     {
-        status = Attributes::NodeLabel::Set(endpoint, chip::CharSpan(nodeLabel, strlen(nodeLabel)));
-        VerifyOrdo(EMBER_ZCL_STATUS_SUCCESS == status, ChipLogError(Zcl, "Error setting Node Label: 0x%02x", status));
+        if (aPath.mAttributeId == app::Clusters::Basic::Attributes::NodeLabel::Id)
+        {
+            char nodeLabel[DeviceLayer::ConfigurationManager::kMaxNodeLabelLength + 1];
+            if (ConfigurationMgr().GetNodeLabel(nodeLabel, sizeof(nodeLabel)) == CHIP_NO_ERROR)
+            {
+                CharSpan span(nodeLabel, strlen(nodeLabel));
+                aEncoder.Encode(span);
+            }
+        }
+        else if (aPath.mAttributeId == app::Clusters::Basic::Attributes::Location::Id)
+        {
+            char location[DeviceLayer::ConfigurationManager::kMaxLocationLength + 1];
+            size_t codeLen = 0;
+            if (ConfigurationMgr().GetCountryCode(location, sizeof(location), codeLen) == CHIP_NO_ERROR)
+            {
+                CharSpan span(location, strlen(location));
+                aEncoder.Encode(span);
+            }
+        }
+        else if (aPath.mAttributeId == app::Clusters::Basic::Attributes::LocalConfigDisabled::Id)
+        {
+            bool localConfigDisabled;
+            if (ConfigurationMgr().GetLocalConfigDisabled(localConfigDisabled) == CHIP_NO_ERROR)
+            {
+                aEncoder.Encode(localConfigDisabled);
+            }
+        }
+
+        return CHIP_NO_ERROR;
     }
 
-    char location[DeviceLayer::ConfigurationManager::kMaxLocationLength + 1];
-    size_t codeLen = 0;
-    if (ConfigurationMgr().GetCountryCode(location, sizeof(location), codeLen) == CHIP_NO_ERROR)
+    CHIP_ERROR Write(const app::ConcreteDataAttributePath & aPath, app::AttributeValueDecoder & aDecoder) override
     {
-        status = Attributes::Location::Set(endpoint, chip::CharSpan(location, strlen(location)));
-        VerifyOrdo(EMBER_ZCL_STATUS_SUCCESS == status, ChipLogError(Zcl, "Error setting Location: 0x%02x", status));
+        CHIP_ERROR err;
+        if (aPath.mAttributeId == app::Clusters::Basic::Attributes::NodeLabel::Id)
+        {
+            CharSpan span;
+            err = aDecoder.Decode(span);
+            if (err == CHIP_NO_ERROR)
+            {
+                ConfigurationMgr().StoreNodeLabel(span.data(), span.size());
```

Comment:
Does this need to notify reporting that the value is dirty?

More generally, what is the benefit of storing this, and the other attributes here, in the configuration manager instead of the attribute store?  And if we did that, we would get the dirty-marking for free, right?

Reply:
As I know and test, in the attribute store the value is only saved in memory, so when the system reboot, these values will be lost. 

For example, In my understanding, the NodeLabel is a node label that the end-user mark the device. so the only reason this value is changed is set by a controller.  And in this case, I think the node should save them in flash or filesystem. So that next time after the node reboot, all controllers should also get the same value.

The Spec also says, NodeLabel attribute SHOULD be set during initial commissioning and MAY be updated by further reconfigurations.

Reply:
> in the attribute store the value is only saved in memory

That's true so far, but we need to fix that so that setting the storage to "NVM" in ZAP actually works and stores in non-volatile memory.  We don't want to force attribute access interface bits for all the simple values that need to be non-volatile...

Now if we absolutely have to get persistence here before the attribute store part is fixed, then I guess we do this, but then we'll just want to revert it once that part is fixed.

Reply:
> Does this need to notify reporting that the value is dirty?

Thinking about this some more, we shoudl handle that higher up the callstack, so no need to do it here.

But the general question of whether these attributes should use AttributeAccessInterface at all remains.

Reply:
> That's true so far, but we need to fix that so that setting the storage to "NVM" in ZAP actually works and stores in non-volatile memory. 

For NodeLabel and Location, I think it is good. But for LocalConfigDisabled, seems it is not only storage.

Reply:
> > Does this need to notify reporting that the value is dirty?
> 
> Thinking about this some more, we shoudl handle that higher up the callstack, so no need to do it here.
> 
> But the general question of whether these attributes should use AttributeAccessInterface at all remains.

higher up the call stack? do you mean at example/xxx-app? Or here we are calling platform is also up them to app layer?

Reply:
What is the "not only storage" bit for LocalConfigDisabled?

By "higher up the callstack" I mean in the IM code.  I'll make sure that part works right so you don't have to worry about dirty-marking in here.